modules/base/rotation/fixedrotation.cpp

```diff
@@ -186,98 +188,59 @@ namespace {
         "only needed if any of the three axis uses the Object type. In this case, the "
         "location of the attached node is required to compute the relative direction."
     };
+
+    struct [[codegen::Dictionary(FixedRotation)]] Parameters {
+        // This value specifies the direction of the new X axis. If this value is not
+        // specified, it will be computed by completing a right handed coordinate system
+        // from the Y and Z axis, which must be specified instead. If this value is a 
+        // string, it is interpreted as the identifier of another scenegraph node. If this
+        // value is a 3-vector, it is interpreted as a direction vector
+        std::optional<std::variant<std::string, glm::vec3>> xAxis;
+
+        // [[codegen::verbatim(XAxisOrthogonalVectorInfo.description)]]
+        std::optional<bool> xAxisOrthogonal;
+
+        // [[codegen::verbatim(XAxisInvertObjectInfo.description)]]
+        std::optional<bool> xAxisInvert [[codegen::key("xAxis - InvertObject")]];
+
+        // This value specifies the direction of the new Y axis. If this value is not
+        // specified, it will be computed by completing a right handed coordinate system
+        // from the X and Z axis, which must be specified instead. If this value is a
+        // string, it is interpreted as the identifier of another scenegraph node. If this
+        // value is a 3-vector, it is interpreted as a direction vector
+        std::optional<std::variant<std::string, glm::vec3>> yAxis;
+
+        // [[codegen::verbatim(YAxisOrthogonalVectorInfo.description)]]
+        std::optional<bool> yAxisOrthogonal;
+
+        // [[codegen::verbatim(YAxisInvertObjectInfo.description)]]
+        std::optional<bool> yAxisInvert [[codegen::key("yAxis - InvertObject")]];
+
+        // This value specifies the direction of the new Z axis. If this value is not
+        // specified, it will be computed by completing a right handed coordinate system
+        // from the X and Y axis, which must be specified instead. If this value is a
+        // string, it is interpreted as the identifier of another scenegraph node. If this
+        // value is a 3-vector, it is interpreted as a direction vector
+        std::optional<std::variant<std::string, glm::vec3>> zAxis;
+
+        // [[codegen::verbatim(ZAxisOrthogonalVectorInfo.description)]]
+        std::optional<bool> zAxisOrthogonal;
+
+        // [[codegen::verbatim(ZAxisInvertObjectInfo.description)]]
+        std::optional<bool> zAxisInvert [[codegen::key("zAxis - InvertObject")]];
+
+        // [[codegen::verbatim(AttachedInfo.description)]]
+        std::optional<std::string> attached;
```

Comment:
Yes, there will be an error from the compiler itself in the corresponding codegen file, saying something along the lines of "unrecognized identifier "AttchdInf.description" since the `verbatim` command just literally copies whatever is between the brackets into the argument list

Reply:
OK, that sounds good. Follow-up question: what are the git implications of generated code? Are developers going to have a lot of false changes to go through when they make a code change?

Reply:
The .gitignore file filters out everything that is `*_codegen.cpp` so the generated code files will never show up and won't be able to be committed